src/main/scala/is/hail/utils/richUtils/RichDenseMatrixDouble.scala

```diff
@@ -11,6 +13,20 @@ object RichDenseMatrixDouble {
     else
       Some(DenseMatrix.horzcat(ms: _*))
   }
+  
+  def read(in: DataInputStream): DenseMatrix[Double] = {
+    val rows = in.readInt()
+    val cols = in.readInt()
+    
+    val data = new Array[Double](rows * cols)
+    var i = 0
+    while (i < rows * cols) {
+      data(i) = in.readDouble()
```

Comment:
This makes me die inside a little, but the right thing isn't clear so I'll leave it for when we have a better IO strategy.  I just want to go on the record.